devnet-sdk/system2/l1_chain.go

```diff
@@ -0,0 +1,107 @@
+package system2
+
+import (
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/locks"
+)
+
+// L1ChainID identifies a L1Chain by name and chainID, is type-safe, and can be value-copied and used as map key.
+type L1ChainID idWithChain
+
+func (id L1ChainID) String() string {
+	return idWithChain(id).string("L1Chain")
+}
+
+func (id L1ChainID) MarshalText() ([]byte, error) {
+	return idWithChain(id).marshalText("L1Chain")
+}
+
+func (id *L1ChainID) UnmarshalText(data []byte) error {
+	return (*idWithChain)(id).unmarshalText("L1Chain", data)
+}
```

Comment:
Tiny thing: one day a developer will spend 47 minutes debugging an ID issue after they copy-pasted this code and only changed `"L1Chain"` in two places 🤣 . I would just pull it out into a small private const